integration/jnifuse/fs/src/main/java/alluxio/jnifuse/struct/FuseFileInfo.java

```diff
@@ -11,24 +11,31 @@
 
 package alluxio.jnifuse.struct;
 
+import alluxio.jnifuse.utils.NativeLibraryLoader;
 import jnr.ffi.Runtime;
+import jnr.ffi.Struct;
 
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 
-public class FuseFileInfo extends ru.serce.jnrfuse.struct.FuseFileInfo {
-  private final ByteBuffer buffer;
+public interface FuseFileInfo {
 
-  public FuseFileInfo(Runtime runtime, ByteBuffer buffer) {
-    super(runtime);
-    this.buffer = buffer;
-    // depends on the arch
-    this.buffer.order(ByteOrder.LITTLE_ENDIAN);
-  }
+  Struct.u_int64_t fh();
+
+  Struct.Signed32 flags();
```

Comment:
Making FuseFileInfo an interface is one way to handle the different structures of `struct fuse_file_info` in libfuse2 and libfuse3, but all the access to `fh` and `flags` has to be changed to method invocations (`fh()`), which may need changes on lots of existing code. 

A way to avoid rewriting existing code is to write `FuseFileInfo` as a base class to `Fuse2FuseFileInfo` and `Fuse3FuseFileInfo` and define the `fh` and `flags` fields in the `FuseFileInfo` class. The actual field structure is defined in the constructor as the order to initialize `Struct.xxx` class. However this way, we cannot reuse `ru.serce.jnrfuse.struct.FuseFileInfo` and have to write `Fuse2FuseFileInfo` on the `struct fuse_file_info` in libfuse2 like the `Fuse3FuseFileInfo`.